test/autotest_test.py

```
@@ -1220,6 +1410,83 @@ def test_spec_records_restart_directory_mode(self):
+                (work.ckpt_dir / "dmtcp_env.txt").read_text(
+                    encoding="utf-8"),
+                f"VAR={tmp_path}\n",
+            )
+            self.assertEqual(
+                (work.ckpt_dir / "nested/file.txt").read_text(
+                    encoding="utf-8"),
+                "nested\n",
+            )
+
+    def test_checkpoint_dir_files_reject_escape_paths(self):
+        for bad_path in ("../bad", "/tmp/bad"):
```

Ruff (0.15.18)

[error] Probable insecure usage of temporary file or directory: "/tmp/bad"

(S108)